src/dex/algebra/algebra.ts

```diff
@@ -730,30 +815,46 @@ export class Algebra extends SimpleExchange implements IDex<AlgebraData> {
     data: AlgebraData,
     side: SwapSide,
   ): Promise<SimpleExchangeParam> {
-    const swapFunction =
-      side === SwapSide.SELL
-        ? UniswapV3Functions.exactInput
-        : UniswapV3Functions.exactOutput;
-
-    const path = this._encodePath(data.path, side);
-    const swapFunctionParams: UniswapV3SimpleSwapParams =
-      side === SwapSide.SELL
-        ? {
-            recipient: this.augustusAddress,
-            deadline: getLocalDeadlineAsFriendlyPlaceholder(),
-            amountIn: srcAmount,
-            amountOutMinimum: destAmount,
-            path,
-          }
-        : {
-            recipient: this.augustusAddress,
-            deadline: getLocalDeadlineAsFriendlyPlaceholder(),
-            amountOut: destAmount,
-            amountInMaximum: srcAmount,
-            path,
-          };
+    let swapFunction;
+    let swapParams;
+
+    if (data.feeOnTransfer) {
+      swapFunction = AlgebraFunctions.exactInputWithFeeToken;
```

Comment:
I think it's worth adding an invariant check here to asses that we have exactly 1 pool involved.

So something like 
```javascript
assert(data.path.length != 1, 'LOGIC ERROR: multihop not supported for feeOnTransfer token');
```